File: front-end/src/stores/types/dashboardConfig.ts
import { LocalizableValue } from "../../schemas";
import { DashboardEditType } from "../../schemas/dashboardEdit";
import { BaseState } from "./base";

export type DashboardConfig = {
  title: LocalizableValue;
  enableOutagesSchedule: boolean;
  outagesScheduleQueue: string;
};

export type DashboardConfigState = BaseState & {
  config?: DashboardConfig;
  editingConfig?: DashboardEditType;
  changed: boolean;
};

File: shared/python/shared/models/dashboard_config.py
from typing import Optional
from beanie import Document

from .localizable_value import LocalizableValue


class DashboardConfig(Document):
    title: LocalizableValue
    enable_outages_schedule: bool = False
    outages_schedule_queue: Optional[str]

    class Settings:
        name = "dashboard_config"
